docs/design/import.md

```diff
@@ -0,0 +1,314 @@
+    State: schema.ImportStatePassthrough,
+},
+```
+
+Otherwise, custom provider implementations were required.
+
+## Caveats
+
+Terraform supports the ability to import multiple resource states during a single import (for example, resource import of an `aws_s3_bucket` also imports any associated `aws_s3_bucket_policy`), however almost all resources in the Terraform ecosystem with import support implement a 1:1 import. The extremely uncommon multiple resource support in practice is likely due to:
+
+- Risk of resource destruction for incorrect implementations ([example issue reference](https://github.com/hashicorp/terraform-provider-aws/issues/6036)).
+- Pracitioners may not have required permissions or wish to manage secondary imported resources ([example issue reference](https://github.com/hashicorp/terraform-provider-aws/issues/9508)).
+- Providers must define the resource addresses for secondary resources, which may not match practitioner desires and require additional action such as `state mv` ([example issue reference](https://github.com/hashicorp/terraform-provider-aws/issues/9001)).
+- (Previously) Risk of resource destruction for missing configurations ([example issue reference](https://github.com/hashicorp/terraform-provider-aws/issues/9001)).
+- (Previously) Terraform CLI unable to refresh secondary resource states using the correct provider instance ([example issue reference](https://github.com/hashicorp/terraform-provider-aws/issues/394)).
+
+While the framework design should not necessarily prohibit the inclusion of this functionality, enabling it with the current Terraform CLI handling could further contribute to provider and practitioner confusion.
+
+## Goals
+
+Import support in this framework should be:
+
+- Able to support diagnostics and cancellation contexts.
+- Available as exported functionality for provider developers.
+- Abstracted from terraform-plugin-go and convertable into those types to separate implementation concerns.
+- Ergonomic to implement Go code (e.g. have helper methods for common use cases).
+
+Additional consideration should be given to:
+
+- Whether implementations must return a partial or full state, since Terraform CLI currently implements a full resource refresh.
+- Accessing the provider instance (e.g. imports that may require additional remote information).
+- Returning multiple resource states.
+- Potential future `terraform import` capabilities (e.g. configuration-based import).
+
+## Proposals
+
+These proposals are split into sections based on separate design considerations.
+
+### Defining Resource Import Support
+
+#### ResourceType Interface
+
+The framework can provide an extension interface type on the existing `ResourceType`:
+
+```go
+// ResourceTypeWithImportState represents a resource type with import support.
+type ResourceTypeWithImportState interface {
+    ResourceType
+
+    ImportState(context.Context, ImportResourceStateRequest, *ImportResourceStateResponse)
```

Comment:
I think it's worth weighing the pros/cons of making import required.

Reply:
It didn't actually take too much convincing myself, but I actually really like the idea of requiring implementation on `Resource` assuming we offer an easy opt-out (e.g. return error diagnostic helper). I have added a new commit with this new proposal idea and recommending it. 🥲